### ISSUE TITLE
tf.sum returns incorrect result on some android devices

### ISSUE THREAD

**Issue by pvanhaes** [tensorflow/tfjs#2836]
### TensorFlow.js version
`1.6.0`

#### Browser version 
Device: Samsung Galaxy S9
Chrome for Android version 80
Firefox 68
Samsung internet 11.1

Also reproduced on the following devices that I had available:
Honor 9X with default browser
Samsung Galaxy S8+ with Chrome

#### Describe the problem or feature request
I have noticed a nasty error when running my model on my Samsung Galaxy S9 with any of the browsers I have installed.
I originally noticed that the output of a softmax operations did not sum to one but instead to around 0.111.
When trying to reduce the scope of this problem I managed to see that the sum operation does not return the expected result depending on the shape of the input tensor.
This only happens with the webgl backend.

#### Code to reproduce the bug / link to feature request
```js
let i = 1;
while (i <= 100) {
    const s = tf.sum(tf.ones([i])).arraySync();
    if (Math.abs(s-i) > 0.1)
        console.log(`returned ${s} instead of ${i}`);
    i += 1;
}
```
Output on my Galaxy S9:
```
returned 36 instead of 28
returned 37 instead of 29
returned 38 instead of 30
returned 39 instead of 31
returned 45 instead of 37
returned 55 instead of 47
returned 61 instead of 53
returned 74 instead of 58
returned 69 instead of 61
returned 78 instead of 62
returned 79 instead of 71
returned 90 instead of 74
returned 111 instead of 87
returned 117 instead of 93
returned 110 instead of 94
```
This problem does not happen on my laptop (Manjaro, Chrome 80).

GitHub issues for this repository are tracked in the [tfjs union repository](https://github.com/tensorflow/tfjs/issues).

Please file your issue there, following the guidance in [that issue template](https://github.com/tensorflow/tfjs/blob/master/ISSUE_TEMPLATE.md).

**Comment by rthadur** [comment]
@annxingyuan i don't have android device to test this , is there any way you can reproduce ? Thank you

**Comment by pvanhaes** [comment]
Did anyone manage to reproduce this issue?
This impacts the last operation of our model which is a softmax, so for now we compute the softmax in plain javascript to avoid this problem.

**Comment by gaikwadrahul8** [comment]
Hi, @pvanhaes 

Apologize for the delayed response and we're re-visiting our older issues and checking whether those issues got resolved or not as of now so May I know are you still looking for the solution or your issue got resolved ? 

If issue still persists after trying with latest version of TFJs please let us know with error log and code snippet to replicate the same issue from our end ?

Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!
